app/uk/gov/hmrc/gform/submission/SubmissionService.scala

```diff
@@ -228,4 +230,13 @@ object SubmissionServiceHelper {
     sectionsToSubmit.traverse(toSectionFormField)
   }
 
+  def getEmailParameterValues(formTemplate: FormTemplate, form: Form): Map[String, String] =
+    formTemplate.emailParameters
+      .flatMap(
+        parameter =>
+          form.formData.fields
+            .find(field => field.id.value == parameter.emailTemplateVariable)
```

Comment:
This looks up field values using the emailTemplateVariable, but that read ${xxx}, but you need to search for xxx not ${xxx}, see https://confluence.tools.tax.service.gov.uk/display/DF/05.+Gforms+Specification